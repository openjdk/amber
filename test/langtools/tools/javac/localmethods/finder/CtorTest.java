/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/fail/ref=CtorTest.out -Werror -XDrawDiagnostics -XDfind=local-methods CtorTest.java
 */

public class CtorTest {

    private static int foo() { // can be local.
        return 42;
    }

    private void goo() {} // can be local.

    private CtorTest(int i) { goo(); } // never flag a constructor.

    public static void main(String [] args) {
        new CtorTest(foo());
    }
}
