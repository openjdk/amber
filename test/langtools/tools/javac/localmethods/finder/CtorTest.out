CtorTest.java:9:24: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, CtorTest
CtorTest.java:13:18: compiler.warn.method.could.be.declared.local: CtorTest(int), kindname.class, CtorTest
- compiler.err.warnings.and.werror
1 error
2 warnings
